modules/unsupported/process-feature/src/main/java/org/geotools/process/vector/GroupCandidateSelectionProcess.java

```diff
@@ -298,15 +297,21 @@ public boolean hasNext() {
             Feature bestFeature = null;
             while (super.hasNext()) {
                 Feature f = super.next();
+                Comparable operationValue = getComparableFromEvaluation(f);
                 if (bestFeature == null) {
-                    // no features in the list this is the first of the group
-                    // takes the values to check the following features if belong to the same group
-                    groupingValues = getGroupingValues(groupingValues, f);
-                    bestFeature = f;
+                    // if null skipping
+                    if (operationValue != null) {
```

Comment:
if (bestFeature == null && operationValue != null) ?

Reply:
The condition is splitted to avoid entering the else if operationValue is Null and bestFeature too. Anyway I've modified the code for better readability.